src/curve_client_tools.hpp

```diff
@@ -146,9 +146,6 @@ struct curve_client_tools_t
                                  const uint8_t *metadata_plaintext,
                                  const size_t metadata_length)
     {
-        const size_t max_metadata_length = 256;
-        zmq_assert (metadata_length < max_metadata_length);
```

Comment:
At this point, yes. I inserted this check as part of my previous PR #2680. Sorry, I did not reflect that properly in the description of this PR. Still, the assertion could be triggered by correct code using libzmq, so this needs to be changed.